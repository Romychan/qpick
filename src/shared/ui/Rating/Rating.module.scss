@import '~/app/assets/styles/_typography.scss';

.container {
  display: flex;
  align-items: center;
  gap: 0.33rem;
}

.icon {
  aspect-ratio: 1/1;
  fill: var(--yellow-300);
  transform: translateY(-1px);
}

.text {
  @include text-md;
  @include text-semibold;
  color: var(--gray-600);
}
