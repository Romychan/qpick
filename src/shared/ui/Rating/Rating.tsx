import { Icon } from '../Icon';

import styles from './Rating.module.scss';

interface IRatingProps {
  /** The value of the rating score */
  value: number;
}

/**
 * This component is used to display the rating of something.
 * For example, `<Rating />` can be placed at the product card to display user ratings.
 */
export const Rating = ({ value }: IRatingProps) => {
  return (
    <div className={styles.container} data-testid="rating">
      <Icon name="star-filled" size={24} className={styles.icon} />
      <p className={styles.text}>{value}</p>
    </div>
  );
};
