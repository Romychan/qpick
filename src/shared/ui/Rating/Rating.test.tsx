import { RenderResult, render, screen } from '@testing-library/react';

import { Rating } from './Rating';

describe('Rating', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = render(<Rating value={4.1} />);
  });

  it('should render correctly', () => {
    const ratingElement = screen.getByTestId('rating');
    const iconElement = screen.getByTestId('icon');

    expect(ratingElement).toBeInTheDocument();
    expect(iconElement).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
