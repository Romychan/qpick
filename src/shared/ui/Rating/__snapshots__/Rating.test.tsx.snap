// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rating > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_861903"
        data-testid="rating"
      >
        <svg
          class="icon icon-star-filled _icon_861903 filled"
          data-testid="icon"
          height="24"
          width="24"
        >
          <use
            xlink:href="/images/icons.svg#star-filled"
          />
        </svg>
        <p
          class="_text_861903"
        >
          4.1
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_861903"
      data-testid="rating"
    >
      <svg
        class="icon icon-star-filled _icon_861903 filled"
        data-testid="icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="/images/icons.svg#star-filled"
        />
      </svg>
      <p
        class="_text_861903"
      >
        4.1
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
